include/deal.II/base/array_view.h

```diff
@@ -286,6 +349,52 @@ make_array_view (std::vector<ElementType> &vector,
 }
 
 
+/**
+ * Create an ArrayView that takes a pair of iterators as arguments. The type
+ * of the ArrayView is inferred from the value type of the iterator (e.g., the
+ * view created from two const iterators will have a const type).
+ *
+ * @warning This function only makes sense for array or vector iterators
+ * (i.e., the memory between @p begin and @end contains valid objects). This
```

Comment:
I'll rewrite that sentence.

There was a [C++17 proposal](https://stackoverflow.com/questions/35004633/checking-if-a-sequence-container-is-contiguous-in-memory#35008842) for an iterator tag that was ultimately not accepted, so I don't believe we have any way of knowing this.